commit 2a0a63e1bb4627c2f503c43cf524c68565c0b205
Author: norman <norman@apache.org>
Date:   Thu Dec 1 09:39:48 2011 +0100

    Remove the usage of AtomicLong as this only confuse users. The Handler
    is not suited for re-use anyway. See #93

diff --git a/src/main/java/org/jboss/netty/example/discard/DiscardClientHandler.java b/src/main/java/org/jboss/netty/example/discard/DiscardClientHandler.java
index 9a991a8..7d4611d 100644
--- a/src/main/java/org/jboss/netty/example/discard/DiscardClientHandler.java
+++ b/src/main/java/org/jboss/netty/example/discard/DiscardClientHandler.java
@@ -15,7 +15,6 @@
  */
 package org.jboss.netty.example.discard;
 
-import java.util.concurrent.atomic.AtomicLong;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
@@ -44,7 +43,7 @@ public class DiscardClientHandler extends SimpleChannelUpstreamHandler {
     private static final Logger logger = Logger.getLogger(
             DiscardClientHandler.class.getName());
 
-    private final AtomicLong transferredBytes = new AtomicLong();
+    private long transferredBytes = 0;
     private final byte[] content;
 
     public DiscardClientHandler(int messageSize) {
@@ -56,7 +55,7 @@ public class DiscardClientHandler extends SimpleChannelUpstreamHandler {
     }
 
     public long getTransferredBytes() {
-        return transferredBytes.get();
+        return transferredBytes;
     }
 
     @Override
@@ -90,7 +89,7 @@ public class DiscardClientHandler extends SimpleChannelUpstreamHandler {
 
     @Override
     public void writeComplete(ChannelHandlerContext ctx, WriteCompletionEvent e) {
-        transferredBytes.addAndGet(e.getWrittenAmount());
+        transferredBytes =+e.getWrittenAmount();
     }
 
     @Override
diff --git a/src/main/java/org/jboss/netty/example/discard/DiscardServerHandler.java b/src/main/java/org/jboss/netty/example/discard/DiscardServerHandler.java
index 67ec946..224b272 100644
--- a/src/main/java/org/jboss/netty/example/discard/DiscardServerHandler.java
+++ b/src/main/java/org/jboss/netty/example/discard/DiscardServerHandler.java
@@ -15,7 +15,6 @@
  */
 package org.jboss.netty.example.discard;
 
-import java.util.concurrent.atomic.AtomicLong;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
@@ -40,10 +39,10 @@ public class DiscardServerHandler extends SimpleChannelUpstreamHandler {
     private static final Logger logger = Logger.getLogger(
             DiscardServerHandler.class.getName());
 
-    private final AtomicLong transferredBytes = new AtomicLong();
+    private long transferredBytes = 0;
 
     public long getTransferredBytes() {
-        return transferredBytes.get();
+        return transferredBytes;
     }
 
     @Override
@@ -59,7 +58,7 @@ public class DiscardServerHandler extends SimpleChannelUpstreamHandler {
     @Override
     public void messageReceived(ChannelHandlerContext ctx, MessageEvent e) {
         // Discard received data silently by doing nothing.
-        transferredBytes.addAndGet(((ChannelBuffer) e.getMessage()).readableBytes());
+        transferredBytes += (((ChannelBuffer) e.getMessage()).readableBytes());
     }
 
     @Override
